compiler/rustc_type_ir/src/flags.rs

```diff
@@ -128,3 +134,361 @@ bitflags::bitflags! {
         const HAS_BINDER_VARS             = 1 << 23;
     }
 }
+
+#[derive(Debug)]
+pub struct FlagComputation {
+    pub flags: TypeFlags,
+
+    /// see `Ty::outer_exclusive_binder` for details
+    pub outer_exclusive_binder: ty::DebruijnIndex,
+}
+
+impl FlagComputation {
+    fn new() -> FlagComputation {
+        FlagComputation { flags: TypeFlags::empty(), outer_exclusive_binder: ty::INNERMOST }
+    }
+
+    #[allow(rustc::usage_of_ty_tykind)]
+    pub fn for_kind<I: Interner>(kind: &ty::TyKind<I>) -> FlagComputation {
```

Comment:
Instead of making everything generic over interner, could you give `FlagComputation` a `_interner: PhantomData<I>`, so that we only need to turbofish `::<I>` on `FlagComputation::<I>::new`? 

It feels a lot less invasive that way.